### PR TITLE
[WIP, don't merge!] Fix check for unclaimed disks (bnc#916442)

### PR DESCRIPTION
**Pull request by toabctl**
While checking if disks are already claimed, don't use the unique
name. Instead use the current real device name (i.e. /dev/sda). This
name can change but then it changes also for the links we check so this
should work.

### CODE REVIEW

File: chef/cookbooks/barclamp/libraries/barclamp_library.rb [hunk 1]
```diff
@@ -383,9 +383,22 @@ def unique_name
...
+                             "owner: #{v[:owner]})")
+              return true
+            end
+          end
+
+          begin
+            not @node[:crowbar_wall][:claimed_disks][self.unique_name][:owner].to_s.empty?
+          rescue
+            false
+          end
```
Comment (vuntz) [inline chef/cookbooks/barclamp/libraries/barclamp_library.rb]: Do we still need the old code?

Comment (toabctl) [inline chef/cookbooks/barclamp/libraries/barclamp_library.rb]: > Do we still need the old code?

@vuntz Good question: tbh I'm not sure howto handle an empty ```owner``` in a claimed disk. Is that really possible that a disk was claimed but has no owner?

Comment (vuntz) [inline chef/cookbooks/barclamp/libraries/barclamp_library.rb]: Well, you have that in your new code, don't you? So it shouldn't matter.

But to answer your question: I think in the past, unclaiming was just making the owner empty. That's why the check is like this.

Comment (toabctl) [inline chef/cookbooks/barclamp/libraries/barclamp_library.rb]: > Well, you have that in your new code, don't you? So it shouldn't matter.

Oh. Right. I played with that and forgot that I readded it. So the original code can be removed.

> But to answer your question: I think in the past, unclaiming was just making the owner empty. That's why the check is like this.

Hm. Sounds strange. So a Mash with claimed disks can have unclaimed disk. Great...

Comment (toabctl) [inline chef/cookbooks/barclamp/libraries/barclamp_library.rb]: > Well, you have that in your new code, don't you? So it shouldn't matter.

@vuntz Fixed in https://github.com/toabctl/barclamp-deployer/commit/ec18aba11adcf4231b02dfd3819ff39a1f8aa20c

**Comment by vuntz** [comment]
I think I'm okay with the patch, but feel a bit uncomfortable merging it right now.

**Comment by toabctl** [comment]
@vuntz: What do you think about https://github.com/crowbar/barclamp-deployer/pull/236 ? It's an alternative approach to solve the problem. I'm somehow undecided yet but currently I think https://github.com/crowbar/barclamp-deployer/pull/236 is the better approach...

File: chef/cookbooks/barclamp/libraries/barclamp_library.rb [hunk 2]
```diff
@@ -383,8 +383,17 @@ def unique_name
         end
 
         def claimed?
-          not @node[:crowbar_wall][:claimed_disks][self.unique_name][:owner].to_s.empty?
-        rescue
+          claimed_disks = @node[:crowbar_wall][:claimed_disks].to_hash
+          claimed_disks.each do | claimed_name, v |
```
Comment (tboerger) [inline chef/cookbooks/barclamp/libraries/barclamp_library.rb]: No spaces after ```|``` and before ```|```

**Comment by toabctl** [comment]
https://github.com/crowbar/barclamp-deployer/pull/236 seems to be the better approach.

**Pull request closed without merging**